flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphCoLocationRestartTest.java

```diff
@@ -130,22 +132,23 @@ public void testConstraintsAfterRestart() throws Exception {
 			timeout);
 
 		//checking execution vertex properties
-		validateConstraints(eg);
+		validateConstraints(eg, slotSharingStrategyFactory.getMostRecentStrategyInstance());
 
 		ExecutionGraphTestUtils.finishAllVertices(eg);
 
 		assertThat(eg.getState(), is(FINISHED));
 	}
 
-	private void validateConstraints(ExecutionGraph eg) {
+	private void validateConstraints(ExecutionGraph eg, SlotSharingStrategy slotSharingStrategy) {
 
 		ExecutionJobVertex[] tasks = eg.getAllVertices().values().toArray(new ExecutionJobVertex[2]);
 
 		for (int i = 0; i < NUM_TASKS; i++) {
-			CoLocationConstraint constr1 = tasks[0].getTaskVertices()[i].getLocationConstraint();
-			CoLocationConstraint constr2 = tasks[1].getTaskVertices()[i].getLocationConstraint();
-			assertThat(constr1.isAssigned(), is(true));
-			assertThat(constr1.getLocation(), equalTo(constr2.getLocation()));
+			ExecutionVertexID executionVertexID0 = tasks[0].getTaskVertices()[i].getID();
+			ExecutionVertexID executionVertexID1 = tasks[1].getTaskVertices()[i].getID();
+
+			assertThat(slotSharingStrategy.getExecutionSlotSharingGroup(executionVertexID0).getExecutionVertexIds(), hasItem(executionVertexID1));
+			assertThat(slotSharingStrategy.getExecutionSlotSharingGroup(executionVertexID1).getExecutionVertexIds(), hasItem(executionVertexID0));
```

Comment:
Instead of relying on internal details, I think it would be good enough to assert the following:

```
final TaskManagerLocation location0 = tasks[0].getTaskVertices()[i]
				.getCurrentAssignedResourceLocation();
			final TaskManagerLocation location1 = tasks[1].getTaskVertices()[i]
				.getCurrentAssignedResourceLocation();

			assertThat(location0, is(equalTo(location1)));
```

Moreover, we should set the test up in a way that every `PhysicalSlot` has a different `TaskManagerLocation`. That way we ensure that the colocation constraints are honored.

Reply:
Great. That fixes the visibility changes 👍